elide-datastore/elide-datastore-search/src/main/java/com/yahoo/elide/datastores/search/SearchDataStore.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.datastores.search;
+
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.EntityDictionary;
+
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.Search;
+
+import java.io.File;
+
+import javax.persistence.EntityManagerFactory;
+
+/**
+ * Performs full text search when it can.  Otherwise delegates to a wrapped store.
+ */
+public class SearchDataStore implements DataStore {
+
+   private DataStore wrapped;
+   private EntityDictionary dictionary;
+   private EntityManagerFactory entityManagerFactory;
+   private boolean indexOnStartup = false;
+
+   public SearchDataStore(DataStore wrapped, EntityManagerFactory entityManagerFactory, boolean indexOnStartup) {
+       this.wrapped = wrapped;
+       this.entityManagerFactory = entityManagerFactory;
+       this.indexOnStartup = indexOnStartup;
+
+       String lucenePath = (String) entityManagerFactory
+               .getProperties()
+               .getOrDefault("hibernate.search.default.indexBase", "/tmp/lucene");
+
+       File file = new File(lucenePath);
+       file.mkdirs();
+   }
+
+   @Override
+   public void populateEntityDictionary(EntityDictionary entityDictionary) {
+       wrapped.populateEntityDictionary(entityDictionary);
+
+       if (indexOnStartup) {
+
+           FullTextEntityManager em = Search.getFullTextEntityManager(entityManagerFactory.createEntityManager());
+           try {
+               for (Class<?> entityClass : entityDictionary.getBindings()) {
+                   if (entityDictionary.getAnnotation(entityClass, Indexed.class) != null) {
+                       em.createIndexer(entityClass).startAndWait();
```

Comment:
What is the behavior of waiting? Will elide return a 503 Service Unavailable when waiting?

Reply:
Assuming you are using elide-standalone (Jetty), this happens inside `server.start()`.  This means the server hasn't yet opened a port for new connections.

The client would get a connection refused error:
```Failed to connect to localhost port 8080: Connection refused```